### PR TITLE
Maid-1112 new put flow

### PR DESCRIPTION
**Pull request by maqi**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/284)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/node_interface.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@ pub enum MethodCall {
     Get { type_id: u64, name: NameType, },
     Post,
     Refresh { content: Box<Sendable>, },
+    PutResponse { destination: NameType, payload: Vec<u8> },
```
Comment (benjaminbollen) [inline src/node_interface.rs]: Looks good to me; routing can cater for this

Comment (benjaminbollen) [inline src/node_interface.rs]: Can we rename this though to 'SendOn { destination : NameType }`;  A PutResponse is only generated after conversion from a Put on Error or Reply;  So you can add an error with with payload Vec<u8> to have this payload included.

Comment (maqi) [inline src/node_interface.rs]: OK, if routing  guarantees the payload will be attached to the PutResponse msg.

**Comment by benjaminbollen** [comment]
I'll do it myself

**Pull request merged**